src/others/test/lv_test_display.c

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file lv_test_indev.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_test_display.h"
+#if LV_USE_TEST
+
+#include "../../core/lv_global.h"
+#include "../../lvgl_private.h"
+#include <stdlib.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void dummy_flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * color_p);
+
+static void buf_changed_event_cb(lv_event_t * e);
+static void delete_event_cb(lv_event_t * e);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+#define _state LV_GLOBAL_DEFAULT()->test_state
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_display_t * lv_test_display_create(int32_t hor_res, int32_t ver_res)
+{
+
+    lv_display_t * disp = lv_display_create(hor_res, ver_res);
+    lv_display_set_color_format(disp, LV_COLOR_FORMAT_XRGB8888);
+
+    size_t buf_size = 4 * (hor_res + LV_DRAW_BUF_STRIDE_ALIGN - 1) * ver_res + LV_DRAW_BUF_ALIGN;
+    uint8_t * buf = malloc(buf_size);
```

Comment:
Eventhough the environment is not "resource constained" other process may corrupt the heap of current process in unintened way so asserting here can save the user to debug the debugger :)